base/string_utils.cpp

```diff
@@ -259,6 +259,21 @@ bool EndsWith(std::string const & s1, std::string const & s2)
   return s1.size() >= s2.size() && s1.compare(s1.size() - s2.size(), s2.size(), s2) == 0;
 }
 
+bool EndsWith(strings::UniString const & s1, std::string const & s2)
```

Comment:
Sorry, what? How it's possible to compare characters of UniString (utf32) with characters of std string (utf8 in the best case)???